syft/pkg/apk_metadata.go

```diff
@@ -32,7 +32,8 @@ type ApkMetadata struct {
 	Description      string          `mapstructure:"T" json:"description"`
 	Size             int             `mapstructure:"S" json:"size" cyclonedx:"size"`
 	InstalledSize    int             `mapstructure:"I" json:"installedSize" cyclonedx:"installedSize"`
-	PullDependencies string          `mapstructure:"D" json:"pullDependencies" cyclonedx:"pullDependencies"`
+	PullDependencies []string        `mapstructure:"D" json:"pullDependencies" cyclonedx:"pullDependencies"`
```

Comment:
The nit here is that this ends up being a schema change, so I think we'll need to publish a new schema.

Reply:
yup, that's correct, this is a breaking change to the schema -- to V4 we go! There are instructions of how to create a new schema and bump it in `schema/json/README.md`. I think we should also add a test that asserts we can decode examples of APK metadata from a syft JSON with the format decoders without issue (decoding V4 works in the case with APK metadata).